### PR TITLE
Dih/notification updates

### PR DESCRIPTION
**Pull request by sushilraje**

### CODE REVIEW

File: device-telemetry/app/com/microsoft/azure/iotsolutions/devicetelemetry/RecurringTasksAgent/RecurringTasks.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        this.notificationEventProcessorFactory = notificationEventProcessorFactory;
+        CompletableFuture.runAsync(() -> this.run());
+    }
+
+    @Override
+    public void run() {
+        tryMethod();
+    }
+
+    private void tryMethod(){
```
Comment (molly-moen) [inline device-telemetry/app/com/microsoft/azure/iotsolutions/devicetelemetry/RecurringTasksAgent/RecurringTasks.java]: nit: rename something more descriptive

File: device-telemetry/app/com/microsoft/azure/iotsolutions/devicetelemetry/services/notification/Notification.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Override
+    public void setAlarm(AlarmNotificationAsaModel model) {
+        this.alarm = model;
+    }
+
+    public CompletionStage executeAsync() {
+        try {
+            for(ActionAsaModel action : this.alarm.getActions()) {
+                switch (action.getActionType()){
```
Comment (molly-moen) [inline device-telemetry/app/com/microsoft/azure/iotsolutions/devicetelemetry/services/notification/Notification.java]: what happens if the action isn't email?

File: device-telemetry/app/com/microsoft/azure/iotsolutions/devicetelemetry/services/notification/implementation/LogicApp.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+        return wsRequest;
+    }
+
+    private String generateRuleDetailUrl() {
+        return String.format("https://%s.azurewebsites.net/maintenance/rule/%s", this.solutionName, this.ruleId);
+    }
+
+    private ObjectNode generatePayLoad() {
+        String emailContent = String.format("Alarm fired for rule ID %s: %s. Custom message: %s. Alarm detail page: %s", this.ruleId, this.ruleDescription, this.content, this.generateRuleDetailUrl());
```
Comment (molly-moen) [inline device-telemetry/app/com/microsoft/azure/iotsolutions/devicetelemetry/services/notification/implementation/LogicApp.java]: nit: split this into multiple lines

File: device-telemetry/conf/application.conf [hunk 4]
```diff
@@ -98,14 +112,11 @@ play.filters.csrf.header.bypassHeaders {
...
 // Allows web applications to make requests from the browser across different domains
-// play.filters.enabled += play.filters.cors.CORSFilter
-//play.filters.cors {
-//  pathPrefixes = ["/some/path", ...]
-//  allowedOrigins = ["https://www.contoso.com", ...]
-//  allowedHttpMethods = ["GET", "POST", "PATCH", "PUT", "DELETE"]
-//  allowedHttpHeaders = ["Accept"]
-//  preflightMaxAge = 3 days
-//}
+play.filters.enabled += play.filters.cors.CORSFilter
```
Comment (molly-moen) [inline device-telemetry/conf/application.conf]: should this still be uncommented?

**Comment by ppathan** [comment]
@molly-moen @jillcary Can this PR be closed now?

**Comment by molly-moen** [comment]
These PR changes have been moved here: https://github.com/Azure/remote-monitoring-services-java/pull/71

**Pull request closed without merging**